pyicloud/services/drive.py

```diff
@@ -42,6 +42,8 @@ def get_node_data(self, node_id):
                 ]
             ),
         )
+        if not request.ok:
+            self.session._raise_error(request.status_code, request.reason)
```

Comment:
`_raise_error` is an internal method on the session. It shouldn't be used outside the session unless we decide to change that scope and make it non protected. I think this either needs a refactor or a different approach.

Reply:
Makes sense. I defined a local `_raise_error` function now.